pinot-segment-local/src/main/java/org/apache/pinot/segment/local/realtime/converter/stats/MutableColumnStatistics.java

```diff
@@ -145,6 +146,8 @@ public boolean isSorted() {
 
     // Iterate over all data to figure out whether or not it's in sorted order
     MutableForwardIndex mutableForwardIndex = (MutableForwardIndex) _dataSource.getForwardIndex();
+    Preconditions.checkState(mutableForwardIndex != null,
```

Comment:
I saw that in some places `UnsupportedOperationException` is thrown and in others the precondition is used. Should we consider unifying them?

Reply:
Thanks for the review!

I've done that on purpose. Where I throw `UnsupportedOperationException` are places where the forward index is allowed to be null due to disabling it. The exception is just trying to indicate that the query isn't supported.

On the other hand, we don't allow disabling the forward index for `Mutable` segments at the moment. So I added the `Precondition` more as an assert that the forward index absolutely should exist otherwise it means something is very wrong.

Let me know if you still think these should be unified or if the above reasoning seems alright.

Reply:
Jackie had a similar comment so I've gone ahead and unified it. I think the exception message is enough to clarify when it's not supported due to a valid disabled forward index and where the forward index shouldn't be null.